app/models/host/managed.rb

```diff
@@ -961,10 +961,14 @@ def assign_hostgroup_attributes(attrs = [])
     attrs.each do |attr|
       next if send(attr).to_i == -1
       value = hostgroup.send("inherited_#{attr}")
-      self.send("#{attr}=", value) unless send(attr).present?
+      assign_hostgroup_attribute attr, value
     end
   end
 
+  def assign_hostgroup_attribute(attr, value)
+    self.send("#{attr}=", value) if new_record? || !send(attr).present?
```

Comment:
Nitpick, does `new_record?` do anything here? For a new record, `send(attr).present?` is always going to be false isn't it?